src/core/meshcop/dataset.cpp

```diff
@@ -433,11 +433,8 @@ ThreadError Dataset::Restore(void)
 
     error = otPlatSettingsGet(mInstance, static_cast<uint16_t>(mType == Tlv::kActiveTimestamp ? kKeyActiveDataset :
```

Comment:
I am not sure if I understand. The check `mType == Tlv::kActiveTimestamp` is to determine which key to use (`ActiveDataSet` vs `PendingDataSet`). 

The change in this PR is that if the `otPlatSettingsGet()` fails to retrieve the dataset content, the `mLength` is set to zero (which basically makes the data set empty).